codalab/objects/work_manager.py

```diff
@@ -175,6 +175,25 @@ def check_bundle_actions(self):
 
         return did_something
 
+    def check_timed_out_bundles(self, update_timeout=60*60*24, creation_timeout=60*60*24*30):
+        '''
+        Filters the list of running bundles by their time, and marks those that
+        have been running for > creation_timeout seconds or have not been
+        updated in > update_timeout seconds as FAILED.
+        Default update_timeout:   1 day
+        Default creation_timeout: 1 month (30 days)
+        '''
+        uuids = self.model.search_bundle_uuids(worksheet_uuid=None, user_id=self.model.root_user_id, \
+            keywords=['state='+','.join([State.RUNNING, State.QUEUED])])
```

Comment:
Oops then that's fine!
On Mon, Dec 14, 2015 at 21:19 Andrew <notifications@github.com> wrote:

> In codalab/objects/work_manager.py
> <https://github.com/codalab/codalab-cli/pull/284#discussion_r47597644>:
>
> > @@ -175,6 +175,25 @@ def check_bundle_actions(self):
> >
> >          return did_something
> >
> > +    def check_timed_out_bundles(self, update_timeout=60*60*24, creation_timeout=60*60*24*30):
> > +        '''
> > +        Filters the list of running bundles by their time, and marks those that
> > +        have been running for > creation_timeout seconds or have not been
> > +        updated in > update_timeout seconds as FAILED.
> > +        Default update_timeout:   1 day
> > +        Default creation_timeout: 1 month (30 days)
> > +        '''
> > +        uuids = self.model.search_bundle_uuids(worksheet_uuid=None, user_id=self.model.root_user_id, \
> > +            keywords=['state='+','.join([State.RUNNING, State.QUEUED])])
>
> Was doing this to be consistent with line 206 below, glad to change it
> though
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/codalab/codalab-cli/pull/284/files#r47597644>.
>